tests/unit/scripts/test_lint.py

```diff
@@ -533,6 +534,24 @@ def test_invalid_tomlfile(tmp_path):
 )
 def test_parse_checks_reference_mode(ref, expect):
     result = parse_checks(['style'], reference=ref)
-    key = [i for i in result.keys()][-1]
+    key = list(result.keys())[-1]
     value = result[key]
     assert expect in value['short']
+
+
+def test_checks_are_sorted():
+    checks = {}
+    for i in range(2):
+        for origin in [STYLE_CHECKS, UPG_CHECKS]:
+            checks[[k for k in origin][i]] = [v for v in origin.values()][i]
+    sorted_checks = sorted(checks.items(), key=sort_checks)
+    result = [i[1]['purpose'] + str(i[1]['index']) for i in sorted_checks]
+    assert result == ['S1', 'S2', 'U1', 'U2']
+
+
+@pytest.mark.parametrize(
+    'test_set', [STYLE_CHECKS, UPG_CHECKS]
+)
+def test_lint_codes_are_sequential(test_set):
+    checks = [i['index'] for i in test_set.values()]
+    assert list(range(1, len(checks) + 1)) == checks
```

Comment:
A Yoda expression, this is
```suggestion
    assert checks == list(range(1, len(checks) + 1))
```